internal/pkg/deploy/cloudformation/stack/workload.go

```diff
@@ -37,24 +37,31 @@ const (
 // RuntimeConfig represents configuration that's defined outside of the manifest file
 // that is needed to create a CloudFormation stack.
 type RuntimeConfig struct {
-	ImageRepoURL      string            // ImageRepoURL is the ECR repository URL the container image should be pushed to.
-	ImageTag          string            // ImageTag is the container image's unique tag.
+	Dockerfile        *DockerfileImage  // Optional. Configuration for Dockerfile image.
 	AddonsTemplateURL string            // Optional. S3 object URL for the addons template.
 	AdditionalTags    map[string]string // AdditionalTags are labels applied to resources in the workload stack.
 }
 
+// DockerfileImage represents configuration about Dockerfile image that is needed to
+// create a CloudFormation stack.
+type DockerfileImage struct {
+	RepoURL  string // RepoURL is the ECR repository URL the container image should be pushed to.
+	ImageTag string // Tag is the container image's unique tag.
+}
+
 type templater interface {
 	Template() (string, error)
 }
 
 // wkld represents a containerized workload running on Amazon ECS.
 // A workload can be a long-running service, an ephemeral task, or a periodic task.
 type wkld struct {
-	name string
-	env  string
-	app  string
-	tc   manifest.TaskConfig
-	rc   RuntimeConfig
+	name     string
+	env      string
+	app      string
+	tc       manifest.TaskConfig
+	rc       RuntimeConfig
+	location *string
```

Comment:
ok! I'm trying to brainstorm some way of consolidating `RuntimeConfig` and `location`. 
What do you think of this? Let me know if you can think of something neater
```go

type location interface {
    Location() string
}

func (i ECRImage) Location() string {
   return fmt.Sprintf("%s:%s", i.RepoURL, i.Tag)
}

type wlkd struct {
   ....
   tc manifest.TaskConfig

   image location  // new
   addonsTemplateURL string            
   tags    map[string]string 
}
```

Then in the `manifest` pkg we can do:
```go
func (i Image) Location() string {
   if s.i == nil {
      return ""
   }
   return s.i
}
```

Finally, in `stack/lb_web_svc.go`:
```go
var img location = envManifest.Location()
if rc.Image != nil {
   img = rc.Image
}
```